homeassistant/components/resolution_center/resolution_center.py

```diff
@@ -0,0 +1,126 @@
+"""The resolution center integration."""
+from __future__ import annotations
+
+import dataclasses
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.integration_platform import (
+    async_process_integration_platforms,
+)
+
+from .const import DOMAIN
+from .issue_registry import async_get as async_get_issue_registry
+from .models import Issue, ResolutionCenterProtocol
+
+
+async def async_process_resolution_center_platforms(hass: HomeAssistant):
+    """Start processing resolution center platforms."""
+    hass.data[DOMAIN]["resolution_center_platforms"] = {}
+
+    await async_process_integration_platforms(
+        hass, DOMAIN, _register_resolution_center_platform
+    )
+
+    return True
+
+
+async def _register_resolution_center_platform(
+    hass: HomeAssistant, integration_domain: str, platform: ResolutionCenterProtocol
+):
+    """Register a resolution center platform."""
+    if integration_domain == DOMAIN:
+        return
+    if not hasattr(platform, "async_fix_issue"):
+        raise HomeAssistantError(f"Invalid resolution center platform {platform}")
+    hass.data[DOMAIN]["resolution_center_platforms"][integration_domain] = platform
+
+
+@callback
+def async_create_issue(
+    hass: HomeAssistant,
+    domain: str,
+    issue_id: str,
+    *,
+    breaks_in_ha_version: str | None = None,
```

Comment:
As this is used in code only, we should maybe just accept AwesomeVersion here directly. (There are more places this comment applies).

```suggestion
    breaks_in_ha_version: AwesomeVersion | None = None,
```

Reply:
I'm not sure there's any benefit in requiring actual `AwesomeVersion` objects and keeping them in memory, but we could validate that the `breaks_in_ha_version` can be parsed by AwesomeVersion with the correct strategy?

Reply:
Commit b736493 validates the passed `breaks_in_ha_version`